.github/workflows/run-e2e-tests.yml

```diff
@@ -141,7 +101,7 @@ jobs:
       - run-unit-tests
```

Comment:
The `needs` columns must be updated to include only the actually required job, eg in this case `build-and-test-docker-image`. Referencing `jobs` that don't exist in the file will cause the workflow to fail.